libraries/rush-lib/src/logic/operations/BuildTimePlugin.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  ICreateOperationsContext,
+  IPhasedCommandPlugin,
+  PhasedCommandHooks
+} from '../../pluginFramework/PhasedCommandHooks';
+import { IExecutionResult } from './IOperationExecutionResult';
+import { OperationStatus } from './OperationStatus';
+
+const PLUGIN_NAME: 'BuildTimePlugin' = 'BuildTimePlugin';
+
+interface ITimelineRecord {
+  startTime: number;
+  endTime: number;
+  durationString: string;
+  name: string;
+  status: OperationStatus;
+}
+
+export interface IBuildTimeRecord {
+  project: string;
+  buildTime: number;
+  status: OperationStatus;
+}
+
+export class BuildTimePlugin implements IPhasedCommandPlugin {
+  private _buildTimes: IBuildTimeRecord[];
+
+  public constructor() {
+    this._buildTimes = [];
+  }
+
+  public apply(hooks: PhasedCommandHooks): IBuildTimeRecord[] {
+    hooks.afterExecuteOperations.tap(
+      PLUGIN_NAME,
+      (result: IExecutionResult, context: ICreateOperationsContext): void => {
+        this._buildTimes = _setBuildTimes(result);
+      }
+    );
+    return this._buildTimes;
```

Comment:
This is a little weird. This function will always return the same empty array. If you want to provide programmatic access to the time data, you would be best served providing a separate API to get the array.

Reply:
I don't directly call this `apply` function for the BuildTimePlugin. I am able to directly call the _setBuildTimes() function and get the time data. How should I write the `apply` function in this case? Should I change it to a `void` return type and get rid of the return statement on line 39 or should I do a larger refactor where I don't implement `BuildTimePlugin` as a `IPhasedCommandPlugin`?

Reply:
I simplified this function and wrote a separate API to get the array.